R/Belgium.R

```diff
@@ -0,0 +1,207 @@
+    #' each level of regions. For Belgium, there are only additional data for
+    #' level 1 regions.
+    level_data_urls = list(
+      "1" = list(
+        "deaths" = "https://epistat.sciensano.be/Data/COVID19BE_MORT.csv"
+      )
+    ),
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "deaths_new"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble tribble
+    set_region_codes = function() {
+      self$codes_lookup$`1` <- tibble::tibble(
+        level_1_region_code = c("BE-BRU", "BE-VLG", "BE-WAL"),
+        level_1_region = c("Brussels", "Flanders", "Wallonia")
+      )
+      self$codes_lookup$`2` <- tibble::tribble(
+        ~level_2_region_code,    ~level_2_region, ~level_1_region_code,
+                    "BE-VAN",        "Antwerpen",      "BE-VLG",
+                    "BE-WBR",    "BrabantWallon",      "BE-WAL",
+                    "BE-WHT",          "Hainaut",      "BE-WAL",
+                    "BE-WLG",       "Li\u00e8ge",      "BE-WAL",
+                    "BE-VLI",          "Limburg",      "BE-VLG",
+                    "BE-WLX",       "Luxembourg",      "BE-WAL",
+                    "BE-WNA",            "Namur",      "BE-WAL",
+                    "BE-VOV",   "OostVlaanderen",      "BE-VLG",
+                    "BE-VBR",    "VlaamsBrabant",      "BE-VLG",
+                    "BE-VWV",   "WestVlaanderen",      "BE-VLG",
+                    "BE-BRU",         "Brussels",      "BE-BRU"
+        )
+    },
+
+    #' @description directs to either level 1 or level 2 processing based on
+    #' request.
+    #' @importFrom dplyr select mutate
+    #' @importFrom lubridate as_date ymd_hms
+    clean_common = function() {
+
+      # vroom fails to load two lines in the main data set
+      # For now, we filter out the broken lines and replace them
+      # with the following data shim
+
+      fixed_lines <- tibble::tribble(
+        ~DATE, ~PROVINCE, ~REGION, ~AGEGROUP, ~SEX, ~CASES,
+        "2020-04-22", "Limburg", "Flanders", "50-59", "F", 10,
+        "2021-02-17", "VlaamsBrabant", "Flanders", "10-19", "M", 12
+      ) %>%
+        mutate(DATE = as.Date(DATE), CASES = as.double(CASES))
+
+      self$data$raw$main_fixed <-
+        self$data$raw$main %>%
+          filter((REGION %in% self$codes_lookup[[1]]$level_1_region
+                   | is.na(REGION))) %>%
+        bind_rows(fixed_lines)
+
+      if (self$level == "1") {
+        self$clean_level_1()
+      } else if (self$level == "2") {
+        self$clean_level_2()
+      }
+    },
+
+    #' @description Region-level Data Cleaning
+    # nolint start
+    #' @importFrom dplyr group_by summarise ungroup full_join tally mutate select rename
+    #' @importFrom tidyr replace_na
+    #' @importFrom lubridate ymd
+    # nolint end
+    clean_level_1 = function() {
+      cases_data <- self$data$raw$main_fixed %>%
+        select(DATE, REGION, CASES) %>%
```

Comment:
this is not a must but there is quite a bit of shared code here. Could an a new method to Belgium to handle it? As is is totally fine though.

Reply:
So I think it would need to be something which would be like
```
clean_and_tally = function(data_tibble, region_cols, data_col) {
}
```
which could then be applied against a list of the tibbles and data columns, with `region_cols = c("REGION")` for level 1 and `region_cols = c("REGION","PROVINCE")` for level 2. I think I can see the shape of it but I honestly think it would take me a while to get working right.

Perhaps this refactor could be for 0.9.1? (Or just once we've got all the existing countries in and are tidying.) It's an interesting coding challenge (at least for me) but I'm likely to add another ten commits along the way...

Reply:
Sure sounds sensible. Very much just a nice to have vs a must have.